api/src/test/java/org/openmrs/module/fhir2/api/translators/impl/ObservationQuantityCodingTranslatorImplTest.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.api.translators.impl;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.assertNull;
+import static org.mockito.Mockito.when;
+import static org.openmrs.module.fhir2.FhirConstants.UCUM_SYSTEM_URI;
+
+import java.io.IOException;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.hl7.fhir.r4.model.Coding;
+import org.hl7.fhir.r4.model.SimpleQuantity;
+import org.hl7.fhir.utilities.CSVReader;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.openmrs.Concept;
+import org.openmrs.ConceptNumeric;
+import org.openmrs.ConceptSource;
+import org.openmrs.module.fhir2.TestFhirSpringConfiguration;
+import org.openmrs.module.fhir2.api.FhirConceptService;
+import org.openmrs.module.fhir2.api.FhirConceptSourceService;
+import org.openmrs.module.fhir2.model.FhirConceptSource;
+import org.openmrs.test.BaseModuleContextSensitiveTest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+
+@ContextConfiguration(classes = TestFhirSpringConfiguration.class, inheritLocations = false)
+public class ObservationQuantityCodingTranslatorImplTest extends BaseModuleContextSensitiveTest {
+	
+	private static final String CONCEPT_UUID = "33fdc8ad-fe4d-499b-93a8-8a991c1d488g";
+	
+	private static final String FHIR_UCUM_COMMON_CODES = "fhir-ucum-common.csv";
+	
+	@Mock
+	private FhirConceptService conceptService;
+	
+	@Mock
+	private FhirConceptSourceService conceptSourceService;
+	
+	@Autowired
+	ObservationQuantityCodingTranslatorImpl quantityCodingTranslator;
+	
+	@Before
+	public void setup() {
+		ConceptTranslatorImpl conceptTranslator = new ConceptTranslatorImpl();
+		conceptTranslator.setConceptService(conceptService);
+		conceptTranslator.setConceptSourceService(conceptSourceService);
+		
+		quantityCodingTranslator.setConceptTranslator(conceptTranslator);
+	}
+	
+	/**
+	 * Tests all common fhir ucum codes specified in https://www.hl7.org/fhir/valueset-ucum-common.html
+	 */
+	@Test
+	public void toFhirResource_shouldTranslateConceptWithCommonUCUMUnitCodes() throws IOException {
+		ConceptNumeric cn = new ConceptNumeric();
+		
+		try (CSVReader csvReader = new CSVReader(
```

Comment:
i'd love to run this with a ParameterizedTest but i think we need to either update to Junit5 or Spring 4.2 which introduces the SpringClassRule junit4 rule

Reply:
Heh... we're kind of stuck in the base package here with OpenMRS 2.0.0 which is Spring 4.1 IIRC and JUnit4. Newer versions provide newer versions of those libraries, but then we'd have to drop support for older versions of the OpenMRS platform. One thought here: do we even need Spring in this test at all? I.e., can we drop the `extends BaseModuleContextSensitiveTest` and `@ContextConfiguration` and just make this a plain JUnit 4 test, in which case we can use the `Parameterized` runner...